contracts/impl/TaskImpl.sol

```diff
@@ -84,7 +84,7 @@ library TaskImpl {
         EnigmaCommon.TaskRecord storage task = state.tasks[_taskId];
         require(task.status == EnigmaCommon.TaskStatus.RecordCreated, 'Invalid task status');
 
-        EnigmaCommon.Worker memory worker = state.workers[msg.sender];
+        EnigmaCommon.Worker memory worker = state.workers[state.operatingToStakingAddresses[msg.sender]];
```

Comment:
@apalepu23 @lacabra Just wondering, does this mean that a given staking address can only have 1 worker?

Reply:
This is correct.

If you think about it, with a stake of X, you have Y possibilities of being chosen.
If you then split that stake into 2, say you have 2 workers with X/2 each, each of them will have Y/2 possibilities of being chosen, which in the aggregate will add up to Y/2 + Y/2 = Y, giving you the same possibilities.

If you want to diversify your stake for fault-tolerance reasons, then you have to set up a second staking address.

Reply:
Alright, makes sense. Thanks for the explaination!